test/Driver/EDMF/helper_funcs/compute_subdomain_statistics.jl

```diff
@@ -0,0 +1,204 @@
+  q_liq = PhasePartition(ts).liq
+  q_ice = PhasePartition(ts).ice
+  # here cloudy and dry are indetical as only one will be used based on the value of cld_frac,
+  # but I define both as in the  quadratuhre options to come they will differ and both will be used
+  ## I need to find out how to initiate cloudy and dry structures 
+  if q_liq+q_ice > 0
+    en_a.cld_frac = 1
+    cloudy.q_tot = en.q_tot
+    cloudy.T     = air_temperature(ts)
+    cloudy.R_m   = gas_constant_air(ts) 
+    cloudy.q_vap = q_vap
+    cloudy.q_liq = q_liq
+    cloudy.q_ice = q_ice
+    dry.q_tot = en.q_tot
+    dry.T     = cloudy.T
+    dry.R_m   = cloudy.R_m
+    dry.q_vap = cloudy.q_vap
+    dry.q_liq = FT(0)
+    dry.q_ice = FT(0)
+  else
+    en_a.cld_frac = 0
+    dry.q_tot = en.q_tot
+    dry.T     = air_temperature(ts)
+    dry.R_m   = gas_constant_air(ts) 
+    dry.q_vap = PhasePartition(ts).vap
+    dry.q_liq = FT(0)
+    dry.q_ice = FT(0)
+    cloudy.q_tot = en.q_tot
+    cloudy.T     = dry.T
+    cloudy.R_m   = dry.R_m
+    cloudy.q_vap = q_vap
+    cloudy.q_liq = q_liq
+    cloudy.q_ice = q_ice
+  end
+  return cloudy, dry
+end
+
+## the complete coding of this function can wait for a working model with SubdomainMean
+function compute_subdomain_statistics!(
+  m::SingleStack{FT,N},
+  state::Vars,
+  aux::Vars,
+  t::Real,
+  statistical_model::LognormalQuadrature,
+  ) where {FT, N}
+
+  gm_a = aux
+  en_a = aux.edmf.environment
+  up_a = aux.edmf.updraft
+  gm = state
+  en = state.edmf.environment
+  up = state.edmf.updraft
+
+  env_len = 10
+  src_len = 6
+
+  inner_env = zeros(env_len);
+  outer_env = zeros(env_len);
+  inner_src = zeros(src_len);
+  outer_src = zeros(src_len);
+  i_Sqt = collect(1:src_len);
+  i_SI = collect(1:src_len);
+  i_Sqt_I = collect(1:src_len);
+  i_Sqt_qt = collect(1:src_len);
+  i_SI_I = collect(1:src_len);
+  i_SI_qt = collect(1:src_len);
+  i_ql = collect(1:env_len);
+  i_T = collect(1:env_len);
+  i_I = collect(1:env_len);
+  i_ρ = collect(1:env_len);
+  i_cf = collect(1:env_len);
+  i_qt_cld = collect(1:env_len);
+  i_qt_dry = collect(1:env_len);
+  i_T_cld = collect(1:env_len);
+  i_T_dry = collect(1:env_len);
+  i_rf = collect(1:env_len);
+
+  if (en.q_tot_cv > eps(FT) && en.e_int_cv > eps(FT) && fabs(en.e_int_q_tot_cv) > eps(FT)
+      && en.q_tot > eps(FT) && sqrt(en.q_tot_cv) < en.q_tot)
+    σ_q = sqrt(log(en.q_tot_cv/en.q_tot/en.q_tot)+1)
+    σ_h = sqrt(log(en.cv_e_int/en.e_int/en.e_int)+1)
+    # Enforce Schwarz's inequality
+    corr = max(min(en.e_int_q_tot_cv/max(σ_h*σ_q, 1e-13),1),-1)
+    sd2_hq = log(corr*sqrt(en.q_tot_cv*en.e_int)/(en.e_int*en.q_tot) + 1.0)
+    σ_cond_θ_q = sqrt(max(σ_h*σ_h - sd2_hq*sd2_hq/σ_q/σ_q, 0.0))
+    μ_q = log(en.q_tot*en.q_tot/sqrt(en.q_tot*en.q_tot + en.q_tot_cv))
+    μ_θ = log(en.e_int*en.e_int/sqrt(en.e_int*en.e_int + en.e_int_cv, k, i]))
+    # clean outer vectors
+    outer_src = FT(0)*outer_src
+    outer_env = FT(0)*outer_env
+
+    for j_qt in 1:m.quadrature_order
```

Comment:
Is there a package we can use to do this? Please document what equations are being solved in this function. Perhaps we can write something simpler here.